src/dataviews/dataviews-factory.js

```diff
@@ -96,6 +96,7 @@ module.exports = Model.extend({
   _generateAttrsForDataview: function (layerModel, attrs, whitelistedAttrs) {
     attrs = _.pick(attrs, whitelistedAttrs);
     attrs.source = attrs.source || { id: layerModel.id };
+    attrs.layerName = attrs.layerName || layerModel.get('layer_name');
```

Comment:
There's a relationship between a DataviewModel and LayerModel...  Why not just add a method to DataviewModel that proxies the request to the layer?


```
// in dataview-model.js...
...
getLayerName: function () {
  return this.layer.get('layer_name');
}
...
```